File: solution/633. Sum of Square Numbers/633_01.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        long long l = 0, r = sqrt(c);
        while (l <= r) {
            long long tmp = l * l + r * r;
            if (tmp == c)
                return true;
            if (tmp < c)
                ++l;
            else
                --r;
        }
        return false;
    }
};

File: solution/633. Sum of Square Numbers/633_01.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        a, b = 0, int(sqrt(c))
        while a <= b:
            m = a*a + b*b
            if m == c:
                return True
            if m < c:
                a += 1
            else:
                b -= 1
        return False
    
